### PR TITLE
Feat/filter and order post find all

### PR DESCRIPTION
**Pull request by Mimachh**
I added some query params for findAll - Post. It works as : post?orderByWhat=likes&orderBy=desc / If there is no params, I've set order by the most recent post. 

I've add some faker post function. 25 post are created with the npm run seed, as role, user, salons.

### CODE REVIEW

File: NewRemarques.md [hunk 1]
```diff
@@ -1,14 +1,3 @@
-## Pinned post : 
-- boolean field on post?
-    If we do that it's hard to validate a unique post pinned by user. The only way would be to get all the post and then request the post where pinned is true and take maybe the lastedUpdatedPost in array. 
-# pinnedPosts = post { where : { pinned: true }} orderBy updatedAt Asc
```
Comment (antikkorps) [inline NewRemarques.md:4]: That is the good request. But it will be better Desc?

File: NewRemarques.md [hunk 2]
```diff
@@ -1,14 +1,3 @@
-## Pinned post : 
-- boolean field on post?
-    If we do that it's hard to validate a unique post pinned by user. The only way would be to get all the post and then request the post where pinned is true and take maybe the lastedUpdatedPost in array. 
-# pinnedPosts = post { where : { pinned: true }} orderBy updatedAt Asc 
-# pinned = pinnedPosts[0];
-
-- table with post id / user id = if user id has already a post in this table, then update (or delete anyway) the old post and update (or create) the new one. Then a user always have an unique pinned...And if we need, to keep an historic of every pinned post's user. This table can be usefull. Instead of deleting the "old" pinned post, we juste create a new one everytime, and take the last entry of each pinned post. Then we keep his historic.
```
Comment (antikkorps) [inline NewRemarques.md:7]: yes that is soft delete... For the user it is deletion put in reality it is saved

File: NewRemarques.md [hunk 3]
```diff
@@ -1,14 +1,3 @@
...
-
-
-L'option pinned tu l'as au moment de la création du post / à l'update. Mais il faut aussi une route spécifique pour ça. Car ça peut être juste un bouton indépendant du reste qui est en hover sur chaque post qui t'appartient.
```
Comment (antikkorps) [inline NewRemarques.md:10]: yes you are right

File: NewRemarques.md [hunk 4]
```diff
@@ -32,10 +21,7 @@ L'option pinned tu l'as au moment de la création du post / à l'update. Mais il
 
 
 
-
-Route d'incrémentation de view / share / repost
 Sur le salon : ajouter un système de recherche par nom / zipcode / country etc.
```
Comment (antikkorps) [inline NewRemarques.md:27]: use an index in Salon I think. I will do it

File: NewRemarques.md [hunk 5]
```diff
@@ -32,10 +21,7 @@ L'option pinned tu l'as au moment de la création du post / à l'update. Mais il
...
 Penser au fonctionnement d'un tag sur un mec. Il faut qu'il recoive une notif de ce tag. Un slug pour le nom et le salon? Pour taguer il faut rechercher.
```
Comment (antikkorps) [inline NewRemarques.md:28]: yes

File: src/post/post.service.ts [hunk 6]
```diff
@@ -8,17 +8,38 @@ import { AuthUserProps } from 'types/all';
...
 
   // In this method we will use the params : pagination, search (user, tag), orbderBy (view, share, likes, comments)
-    async findAll() {
+    async findAll(query: SearchPostDto) {
+      let orderByClause: any = {};
+      const orderByWhat = query.orderByWhat; // ça peut être views / repost
+      const orderBy = query.orderBy;
+
+      if (!orderByWhat || !orderBy) {
+        orderByClause = { createdAt: "desc" };
```
Comment (antikkorps) [inline src/post/post.service.ts:26]: asc? don't you think?

**Review by antikkorps — APPROVED** [review]

File: NewRemarques.md (hunk 1 above)
Comment (Mimachh) [reply NewRemarques.md:4]: I forgot to save my new NewRemarques. This part no longer exist. I've written asc without thinking of. Just to remember to add an orderBy. But yes I think desc would show the latest pinned post.
Should we have a specific route to get the last pinned post u think?

File: NewRemarques.md (hunk 2 above)
Comment (Mimachh) [reply NewRemarques.md:7]: Soft delete is implemented already.

File: NewRemarques.md (hunk 3 above)
Comment (Mimachh) [reply NewRemarques.md:10]: This is done..damn I should save the file lol

File: NewRemarques.md (hunk 4 above)
Comment (Mimachh) [reply NewRemarques.md:27]: Fuck, this is done too..

File: NewRemarques.md (hunk 5 above)
Comment (Mimachh) [reply NewRemarques.md:28]: Ok, that is not done yet.

**Review by Mimachh — COMMENTED** [review]
Sorry I was sur to have save all my files. But my update on New Remarque was not saved.

**Pull request merged**